### PR TITLE
Support Project

### PR DESCRIPTION
**Pull request by bryanjos**
connects to #74 
connects to #106 

**TODO**: 
- JS needed to make rows on fund page dynamic
- JS to disable form button when clicked (to prevent double submission)
- Design updates

#### Description: 
- Add Fund Schema
- Add Pledges Schema
- Migration
- Add Funds Context
- Add Fund Controller
- Add Fund Controller Test
- Add JavaScript code to create escrow account
- Fix lints

screen shot 2018-08-10 at 10 12 49 am

screen shot 2018-08-10 at 10 14 05 am

### CODE REVIEW

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

File: test/incentivize_web/controllers/fund_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+defmodule IncentivizeWeb.RepositoryControllerTest do
```
Comment (lintomat) [inline test/incentivize_web/controllers/fund_controller_test.exs]: Modules should have a @moduledoc tag.

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

File: lib/incentivize/funds/fund.ex [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    |> validate_number(:pledge_amount, greater_than: 0)
+    |> create_stellar_fund()
+  end
+
+  defp create_stellar_fund(changeset) do
+    if get_field(changeset, :supporter_id) do
+      # TODO: make this work
+      # user = Users.get_user(get_field(changeset, :supporter_id))
+      # escrow_public_key = Stellar.create_fund_account(user.stellar_public_key)
+      # changeset = put_change(changeset, :stellar_public_key, escrow_public_key)
```
Comment (bryanjos) [inline lib/incentivize/funds/fund.ex]: @jwietelmann This is the last missing piece for creating the stellar account used for holding supporter funds. I ran into a timeout issue.

```
]}, 5000)
    ** (EXIT) time out
    (elixir) lib/gen_server.ex:924: GenServer.call/3
    (poolboy) src/poolboy.erl:76: :poolboy.transaction/3
    (elixir) lib/task/supervised.ex:89: Task.Supervised.do_apply/2
    (elixir) lib/task/supervised.ex:38: Task.Supervised.reply/5
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Function: #Function<0.15951071/0 in NodeJS.Supervisor.to_transaction/2>
```

File: lib/incentivize/stellar/stellar.ex [hunk 3]
```diff
@@ -12,4 +12,8 @@ defmodule Incentivize.Stellar do
   defp config do
     Confex.get_env(:incentivize, __MODULE__)
   end
+
+  def create_fund_account(supporter_public_key) do
+    NodeJS.call({:repositoryFund, :create}, [network_url(), secret(), supporter_public_key])
```
Comment (bryanjos) [inline lib/incentivize/stellar/stellar.ex]: Here is the code to call into the js to create account

File: nodejs/repositoryFund.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    .build()
+
+  transaction.sign(escrowKeyPair)
+  response = await server.submitTransaction(transaction)
+
+  return response
+}
+
+async function create(network, incentivizeSecret, supporterPublicKey) {
+  const server = createServer(network)
```
Comment (bryanjos) [inline nodejs/repositoryFund.js]: This javascript function creates the account

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/110).

File: assets/css/utilities/_flex.scss [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// Copyright © Revelry Labs, LLC 2017 - All Rights Reserved.
+.u-flexBaseline {
+  @include flex(baseline);
+}
+.u-flexEqualWidthCols {
+  @include breakpoint(medium) {
+    @include flexEqualWidthCols;
```
[the remark this answers is not in the document]
Comment (blazebarsamian) [reply assets/css/utilities/_flex.scss]: this can be ignored

**Comment by blazebarsamian** [comment]
@brittanygay styles look good to me. dont know if the other stuff needs to be reviewed by an engineer before its merged though.

**Pull request merged**